### ISSUE TITLE
What is the preferred software for development?

### ISSUE THREAD

**Issue by cfunseth** [gbfans/afterlife-lightkit#1]
There will likely be library dependencies for the software, and the hardware may require modifications from multiple people. This makes documentation difficult if everyone is using a different environment to develop in.

I'd throw a vote out for just using Arduino and KiCAD since these are both popular and available for anyone to use. 

Just throwing it out there for consideration before design progresses too far.

**Comment by ajquick** [comment]
I would agree that those are my preferences as well. Once you start using VS Code or other software it becomes more complicated for end users to compile their own code.

The PCB was designed in Kicad, so that's what the source will be provided as. I may put up a PDF schematic here soon as I like to test a physical board before I release the actual source files.

**Comment by prodestrian** [comment]
Agree also :+1: 

It's a shame that Arduino IDE doesn't (yet) support `.editorconfig` files:
https://github.com/arduino/Arduino/issues/8804
I still think we'll want to include one for those of us who prefer VSCode, to try to reduce the number of PR's which have to be rejected due to unrelated code-style changes.

Personally I write my code in VSCode but compile in the Arduino IDE (using the "Use External Editor" feature), I've had too many issues with trying to do everything inside VSCode itself. I found Arduino IDE 1.X unusable for writing code, although I haven't yet tried the Arduino IDE 2.0 and I hear it's been greatly improved.
